Model/Sync/Catalog/Processor/Main.php

```diff
@@ -623,4 +623,25 @@ public function setNormalSyncFlag($flag)
     {
         $this->normalSync = $flag;
     }
+
+    /**
+     * @param array <int> $productIds
+     * @param array <int | null> $storeId
+     * @return void
+     */
+    public function removeProductFromSyncTable($productIds, $storeId)
```

Comment:
@manaftryzens Can you please explain why delete the records from the sync table?
We've got status code 000 which indicates that the certain product should be re-synced, which would have the exact same impact.

In the case of deleting records, we would potentially never have observability that there was any attempt to sync the product.

Reply:
If we don't delete it it will first send a PATCH request as the table contains yotpo_id in it evn though it is wrong. So it will send a PATCH then it recieve a 404  , then it will send a GET and PATCH etc..
The benefit of removal action is , it will send a GET first (because there is no yotpo_id) and then a PATCH so the task is complete.